socs/agents/srs_cg635/agent.py

```diff
@@ -0,0 +1,237 @@
+            'frame_length': 10 * 60,
+        }
+        self.agent.register_feed('srs_clock',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    def init(self, session, params=None):
+        """init(auto_acquire=False)
+
+        **Task** - Initialize the connection to the srs clock.
+
+        Parameters
+        ----------
+        auto_acquire: bool, optional
+            Default is False. Starts data acquisition after initialization
+            if True.
+        """
+        with self.lock.acquire_timeout(0) as acquired:
+            if not acquired:
+                return False, "Could not acquire lock"
+
+            try:
+                self.clock = SRSCG635Interface(self.ip_address, self.gpib_slot)
+                self.idn = self.clock.identify()
+
+            except (socket.timeout, OSError) as e:
+                self.log.error(f"Clock timed out during connect: {e}")
+                return False, "Timeout"
+            self.log.info("Connected to Clock: {}".format(self.idn))
+
+        # Start data acquisition if requested in site-config
+        auto_acquire = params.get('auto_acquire', False)
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Initialized Clock.'
+
+    def _initialize_interface(self):
+        """Initialize the SRSCG635 interface."""
+        try:
+            self.clock = SRSCG635Interface(self.ip_address, self.gpib_slot)
+        except (socket.timeout, OSError) as e:
+            self.log.error(f"Clock timed out during connect: {e}")
+            self.clock = None
+            return False
+
+        self.idn = self.clock.identify()
+        self.log.info("Connected to Clock: {}".format(self.idn))
+        self.log.info("Clearing event registers and error queue.")
+        self.clock.clear()
+        return True
+
+    @ocs_agent.param('test_mode', default=False, type=bool)
+    @ocs_agent.param('wait', default=1, type=float)
+    def acq(self, session, params):
+        """acq(wait=1, test_mode=False)
+
+        **Process** - Continuously monitor srs clock lock registers
+        and publish to a feed.
+
+        The ``session.data`` object stores the most recent published values
+        in a dictionary. For example::
+
+            session.data = {
+                'timestamp': 1598626144.5365012,
+                'block_name': 'clock_output',
+                'data': {
+                    'Frequency': 122880000.0
+                    'Standard_CMOS_Output': 3,
+                    'Running_State': 1,
+                    'Timebase': 3,
+                    'PLL_RF_UNLOCKED': 1,
+                    'PLL_19MHZ_UNLOCKED': 1,
+                    'PLL_10MHz_UNLOCKED': 0,
+                    'PLL_RB_UNLOCKED': 1,
+                    'PLL_OUT_UNLOCKED': 0,
+                    'PLL_Phase_Shift': 0,
```

Comment:
Is this the stuff that just measures if the state has changed since it was last queried? I find these fields kinda confusing to parse perhaps we just remove them.